alacritty_terminal/src/grid/resize.rs

```diff
@@ -67,10 +67,12 @@ impl<T: GridCell + Default + PartialEq + Copy> Grid<T> {
             self.scroll_up(&(Line(0)..self.lines), Line(required_scrolling), T::default());
 
             // Clamp cursors to the new viewport size.
-            self.saved_cursor.point.line = min(self.saved_cursor.point.line, target - 1);
             self.cursor.point.line = min(self.cursor.point.line, target - 1);
         }
 
+        // Clamp saved cursor, since only primary cursor is scrolled into viewport.
+        self.saved_cursor.point.line = min(self.saved_cursor.point.line, target - 1);
```

Comment:
Note that this is partially necessary because of #3584, since we do not reflow the cursor ourselves.